log/mcap.ksy

```diff
@@ -0,0 +1,436 @@
+    0x0d: metadata_index
+    0x0e: summary_offset
+    0x0f: data_end
+
+types:
+  magic:
+    seq:
+      - id: magic
+        contents: [0x89, "MCAP0\r\n"]
+
+  prefixed_str:
+    seq:
+      - id: len
+        type: u4
+      - id: str
+        type: str
+        size: len
+        encoding: UTF-8
+
+  tuple_str_str:
+    seq:
+      - id: key
+        type: prefixed_str
+      - id: value
+        type: prefixed_str
+
+  map_str_str:
+    seq:
+      - id: len
+        type: u4
+      - id: entries
+        size: len
+        type: entries
+    types:
+      entries:
+        seq:
+          - id: entries
+            type: tuple_str_str
+            repeat: eos
+
+  records:
+    seq:
+      - id: records
+        type: record
+        repeat: eos
+
+  record:
+    seq:
+      - id: op
+        type: u1
+        enum: opcode
+      - id: len
+        type: u8
+      - id: body
+        size: len
+        type:
+          switch-on: op
+          cases:
+            opcode::header: header
+            opcode::footer: footer
+            opcode::schema: schema
+            opcode::channel: channel
+            opcode::message: message
+            opcode::chunk: chunk
+            opcode::message_index: message_index
+            opcode::chunk_index: chunk_index
+            opcode::attachment: attachment
+            opcode::attachment_index: attachment_index
+            opcode::statistics: statistics
+            opcode::metadata: metadata
+            opcode::metadata_index: metadata_index
+            opcode::summary_offset: summary_offset
+            opcode::data_end: data_end
+
+  header:
+    seq:
+      - id: profile
+        type: prefixed_str
+      - id: library
+        type: prefixed_str
+
+  footer:
+    seq:
+      - id: ofs_summary_section
+        -orig-id: summary_start
+        type: u8
+      - id: ofs_summary_offset_section
+        -orig-id: summary_offset_start
+        type: u8
+      - id: summary_crc32
+        -orig-id: summary_crc
+        type: u4
+        doc: |
+          A CRC32 of all bytes from the start of the Summary section up through and
+          including the end of the previous field (summary_offset_start) in the footer
+          record. A value of 0 indicates the CRC32 is not available.
+    instances:
+      summary_section:
+        io: _root._io
+        pos: ofs_summary_section
+        size: "(ofs_summary_offset_section != 0 ? ofs_summary_offset_section : _root.ofs_footer) - ofs_summary_section"
+        type: records
+        if: ofs_summary_section != 0
+      summary_offset_section:
+        io: _root._io
+        pos: ofs_summary_offset_section
+        size: "_root.ofs_footer - ofs_summary_offset_section"
+        type: records
+        if: ofs_summary_offset_section != 0
+      summary_crc32_input:
+        io: _root._io
+        pos: ofs_summary_section
+        size: "_root._io.size - ofs_summary_section - sizeof<magic> - summary_crc32._sizeof"
+        if: ofs_summary_section != 0
```

Comment:
According to your commit https://github.com/foxglove/mcap/commit/44355df51a2f77b764b42bb9b3b0985200a168ad, `summary_crc32` is calculated from _something_ (not sure from what exactly) even if the summary section is empty (`ofs_summary_section == 0` or `summary_start == 0` in original identifiers). Can the implementation of `summary_crc32_input` be extended to this case?

Reply:
Thanks for the reminder, fixed!